drivers/sensor/ms5607/ms5607_spi.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2019 Thomas Schmid <tom@lfence.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+#include "ms5607.h"
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(ms5607);
+
+#ifdef DT_MEAS_MS5607_BUS_SPI
+
+#if defined(DT_INST_0_MEAS_MS5607_CS_GPIOS_CONTROLLER)
+static struct spi_cs_control ms5607_cs_ctrl;
+#endif
+
+#define SPI_CS NULL
+
+static struct spi_config ms5607_spi_conf = {
+	.frequency = DT_INST_0_MEAS_MS5607_SPI_MAX_FREQUENCY,
+	.operation = (SPI_OP_MODE_MASTER | SPI_WORD_SET(8) |
+		      SPI_MODE_CPOL | SPI_MODE_CPHA |
+		      SPI_TRANSFER_MSB |
+		      SPI_LINES_SINGLE),
+	.slave = DT_INST_0_MEAS_MS5607_BASE_ADDRESS,
+	.cs = SPI_CS,
+};
+
+static int ms5607_spi_raw_cmd(const struct ms5607_data *data, u8_t cmd)
+{
+	int err;
+
+	const struct spi_buf buf = {
+		.buf = &cmd,
+		.len = 1,
+	};
+
+	const struct spi_buf_set buf_set = {
+		.buffers = &buf,
+		.count = 1,
+	};
+
+	err = spi_write(data->ms5607_device, &ms5607_spi_conf, &buf_set);
+	if (err < 0) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int ms5607_spi_reset(const struct ms5607_data *data)
+{
+	int err = ms5607_spi_raw_cmd(data, MS5607_CMD_RESET);
+
+	if (err < 0) {
```

Comment:
if (ms5605_spi_raw_cmd())
...

Reply:
can not do that because of conditional code after error check and return of actual error code